sklearn/kernel_approximation.py

```diff
@@ -682,8 +687,36 @@ def fit(self, X, y=None):
         X = self._validate_data(X, accept_sparse="csr")
         check_non_negative(X, "X in AdditiveChi2Sampler.fit")
 
+        if self.sample_interval is None:
+            # See figure 2 c) of "Efficient additive kernels via explicit feature maps"
+            # <http://www.robots.ox.ac.uk/~vedaldi/assets/pubs/vedaldi11efficient.pdf>
+            # A. Vedaldi and A. Zisserman, Pattern Analysis and Machine Intelligence,
+            # 2011
+            if self.sample_steps == 1:
+                self._sample_interval = 0.8
+            elif self.sample_steps == 2:
+                self._sample_interval = 0.5
+            elif self.sample_steps == 3:
+                self._sample_interval = 0.4
+            else:
+                raise ValueError(
+                    "If sample_steps is not in [1, 2, 3],"
+                    " you need to provide sample_interval"
+                )
+        else:
+            self._sample_interval = self.sample_interval
```

Comment:
At `transform` if `self._sample_interval` does not exist, it means that we did not call `fit`. Therefore, we need to run the exact same code to compute a local `sample_interval`

Reply:
Thank you for your detailed feedback. Just to make sure I understand your last sentence correctly, what you suggest is duplicating the logic in both fit and transform (without the self). I'm not familiar with deprecation cycles yet :)

Reply:
Ups a wrong review here. Somehow I had the impression to have read

```python
self.sample_interval = self._sample_interval
```

Brain freeze only. You can discard the comment :)

Reply:
resolving this one then :)